### ISSUE TITLE
cache.rs looks like it discards errors and extensions

### ISSUE THREAD

**Issue by BrynCooke** [apollographql/router#3404]
Looking at the code in `cache.rs` `cache_call` it looks like errors and extensions are:
1. Not cached
2. Not propagated

It is possible that an entity is returned from a subgraph service with errors (with or without path) so we may need to revisit the design of entity level caching in general. At the very least we need to pass through errors and extensions when we have them.

**Comment by Geal** [comment]
yes, there should be a similar error dispatching to what is done at the client response side
